### PR TITLE
Reset the media store state if collection state changes

### LINKED ISSUE #3976: Staging: Provider filter is preserved on collection tag views
## Description

Part of the #410 project. New collection pages are preserving search filters when they shouldn't. For example, searching under a specific provider, selecting a single result, and clicking on a tag leads that tag view to only include results from the provider instead of all results.

This may apply to other filters that I am not aware of!

## Reproduction

1. Make sure the `additional_search_views` feature is enabled: https://staging.openverse.org/preferences
1. Search "rabbit" from StockSnap only: https://staging.openverse.org/search/image?q=rabbit&source=stocksnap
2. Click on a single result, then a tag
4. Observe that the tag view is filtered to only show StockSnap results. You can refresh and see different results.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #3976 by @zackkrida

## Description

This PR adds `clearMedia` call to the search store's function setting the collection state. This makes the `setCollectionState` similar to `setSearchStateFromUrl`. To make sure that we don't clear media when non-collection state params change (for instance, if we add filters to the collection views, or if we add `page` parameter to the frontend URL to improve pagination), we first check if the `collectionParams` are different.

## Testing Instructions

Go through the following steps and check that the linked issue is fixed (step 5):

1. Run the app locally using `just frontend/run dev`
2. Make sure the additional_search_views feature is enabled: http://localhost:8443/preferences
3. Search "rabbit" from StockSnap only: http://localhost:8443/search/image?q=rabbit&source=stocksnap
4. Click on a single result, then a tag
5. Observe that the tag view is _not_ filtered to only show StockSnap results. You can refresh and see the same results.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/src/data/api-service.ts [hunk 1]
```diff
@@ -122,7 +122,16 @@ export const createApiService = ({
...
+    (response) => {
+      const params = new URLSearchParams(response.config.params).toString()
+      console.log(
+        "> API response",
+        new URL(
+          `${response.config.baseURL}${response.config.url}${params ? "?" + params : ""}`
+        ).href
+      )
+      return response
+    },
```
Comment (obulat) [inline frontend/src/data/api-service.ts:134]:
```suggestion
    (response) => response,
```

**Review by zackkrida — APPROVED** [review]
I don't love having to manage `isCollectionStateSame` as a custom approach to checking for deep equality of two objects, but I am not sure we have a simpler alternative at the moment and this appears to be working. LGTM! We should really move to using lodash or similar in the future IMO.

File: frontend/src/stores/search.ts [hunk 2]
```diff
@@ -333,9 +333,39 @@ export const useSearchStore = defineStore("search", {
 
       this.addRecentSearch(formattedTerm)
     },
+    isCollectionStateSame(
+      collectionParams: CollectionParams,
+      mediaType: SupportedMediaType
+    ) {
```
Comment (AetherUnbound) [inline frontend/src/stores/search.ts:339]: Is there an easy way to test this new function?

**Review by AetherUnbound — COMMENTED** [review]
I wanted to try and review this PR, but it doesn't have any testing instructions so I'm not sure how best to proceed 🙂 I do have one question.

File: frontend/src/stores/search.ts (hunk 2 above)
Comment (obulat) [reply frontend/src/stores/search.ts]: I removed this function in favor of @wordpress/is-shallow-equal that we are already using in another component.

**Comment by obulat** [comment]
> I wanted to try and review this PR, but it doesn't have any testing instructions so I'm not sure how best to proceed 🙂 I do have one question.

Sorry, I updated the instructions :)

**Review by AetherUnbound — APPROVED** [review]
Perfect, thanks for the testing instructions! Glad we can use an existing function here too 😄

**Pull request merged**